### ISSUE TITLE
Incompatible shape of bnorm's param for pretrained resnet models

### ISSUE THREAD

**Issue by peiyunh** [vlfeat/matconvnet#527]
Hi, 

I tested the current `imagenet-resnet-50-dag` model. The bug happens during training, inside the function `accumulate_gradients` of `cnn_train_dag.m`, and due to incompatible shapes between `net.params(i).value` and `net.params(i).der`. 

In pre-trained resnet, both multiplier and bias has shapes of 1x1xK. It works fine with only forward pass. However, for bnorm's backward pass, `vl_nnbnorm` only spits out `derParams` with `ndims=2`. This eventually causes mismatch between `net.params(i).value` and `net.params(i).der` when updating parameters inside `cnn_train_dag.m`. 

I suggest we reshape bnorm's multipliers and biases to be `Kx1` vectors, instead of keeping them `1x1xK`, in the released resnet models. 

Peiyun

**Comment by albanie** [comment]
This is a good suggestion.  As a short term solution until the pretrained networks are modified, this script will reshape the batch normalisation parameters to conform to the shapes expected by the `accumulate_gradients` function:

    % load network
    net = load('imagenet-resnet-50-dag.mat');
    net = dagnn.DagNN.loadobj(net);

    % reshape batch normalization parameters
    for l = 1:numel(net.layers)
        if isa(net.layers(l).block, 'dagnn.BatchNorm')
            paramIdx = net.layers(l).paramIndexes;
            inputs = net.layers(l).inputs;
            numChannels = net.layers(net.getLayerIndex(inputs)).block.size(4);
            for p = paramIdx
                net.params(p).value = reshape(net.params(p).value, numChannels, []);
            end
        end
    end

**Comment by vedaldi** [comment]
Thanks. Fixed in devel.
